translationRecorder/app/src/main/java/org/wycliffeassociates/translationrecorder/TranslationRecorderApp.java

```diff
@@ -40,19 +40,18 @@ public void onCreate() {
         // be used to notify upload status.
         // Gradle automatically generates proper variable as below.
         UploadService.NAMESPACE = BuildConfig.APPLICATION_ID;
-        UploadService.HTTP_STACK = new OkHttpStack(new OkHttpClient());
-
+        UploadService.HTTP_STACK = new OkHttpStack(getOkHttpClient());
     }
 
-    private OkHttpClient getOkHttpClient() throws CertificateException, NoSuchAlgorithmException, KeyStoreException, KeyManagementException, IOException {
+    private OkHttpClient getOkHttpClient() {
         return new OkHttpClient.Builder()
                 .followRedirects(true)
                 .followSslRedirects(true)
                 .retryOnConnectionFailure(true)
-                .connectTimeout(15, TimeUnit.SECONDS)
-                .writeTimeout(30, TimeUnit.SECONDS)
+                .connectTimeout(30, TimeUnit.SECONDS)
+                .writeTimeout(0, TimeUnit.SECONDS)
```

Comment:
any reason why the write timeout is 0?

Reply:
@sarabiaj when it's 0. it means unlimited, because we don't know what kind of connection a user will have, or we can set a very long value (an hour?)

Reply:
oh okay, cool. can you comment "//unlimited timeout"?